itou/www/apply/forms.py

```diff
@@ -436,6 +446,13 @@ def clean_hiring_start_at(self):
             self.add_error("hiring_start_at", forms.ValidationError(JobApplication.ERROR_START_IN_PAST))
         elif hiring_start_at and hiring_start_at > timezone.localdate() + relativedelta(months=6):
             self.add_error("hiring_start_at", forms.ValidationError(JobApplication.ERROR_START_IN_FAR_FUTURE))
+        elif (
+            not self.instance.hiring_without_approval
+            and self.company.is_subject_to_eligibility_rules
+            and self.job_seeker.has_valid_approval
+            and hiring_start_at >= self.job_seeker.latest_approval.end_at
```

Comment:
Le check dans accept utilise une égalité stricte, qui me semble plus judicieuse si on suppose que la date de fin de PASS est inclusive.
Ça mérite un test à mon avis.

La gestion des périodes dans l’application est bancale, et ça me :exploding_head: :sob: 

```suggestion
            and hiring_start_at > self.job_seeker.latest_approval.end_at
```